File: twitterEngine/src/spl/stomp/protocol/DisconnectFrame.java
package spl.stomp.protocol;

/**
 * @name Disconnect Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * DISCONNECT stomp frame
 */
public class DisconnectFrame extends StompFrame {
    public DisconnectFrame() {
        super(StompCommand.DISCONNECT);
    }
    
    public DisconnectFrame(String receipt) {
        this();
        addFrameHeader("receipt", receipt);
    }
    
    @Override
    protected String[] getMandatoryHeaders() {
        return new String[] {"receipt"};
    }
}

File: twitterClient/twitterClient/src/stomp/frames/DisconnectFrame.cpp
//
//  DisconnectFrame.cpp
//  Object to represent the DISCONNECT frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/DisconnectFrame.h"

namespace stomp {
    DisconnectFrame::DisconnectFrame(std::string receipt): StompFrame(DISCONNECT_FRAME_NAME) {
        addHeaderPair("receipt", receipt);
    }
    
    void DisconnectFrame::accept(FrameVisitor *visitor)
    {
    }
}
